Dockerfile

```diff
@@ -13,7 +13,16 @@ RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
 ADD . /eland
 WORKDIR /eland
 
-RUN --mount=type=cache,target=/root/.cache python3 -m pip install torch==1.13.1+cpu --extra-index-url https://download.pytorch.org/whl/cpu
-RUN --mount=type=cache,target=/root/.cache python3 -m pip install --no-cache-dir --disable-pip-version-check .[all]
+ARG TARGETPLATFORM
+RUN --mount=type=cache,target=/root/.cache/pip \
+    if [ "$TARGETPLATFORM" = "linux/amd64" ]; then \
+      python3 -m pip install \
+        --no-cache-dir --disable-pip-version-check --extra-index-url https://download.pytorch.org/whl/cpu  \
```

Comment:
`--no-cache-dir` is a good idea, thanks. That means we should remove `--mount=type=cache,target=/root/.cache/pip` above.

Reply:
I've left the cache mount in because `pip` still leaves something behind in `/root/.cache/pip` even with the `--no-cache-dir` option. 😞